articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+                    <EmailContext.Provider value={email}>
+                      <Email getId={getId} id={id} fav={fav} readList={readList} setReadList={setReadList} />
+                    </EmailContext.Provider>
+                  </div>
+                );
+              })}
+            </Scroll>
+          </div>
+          {dualPanel && emailsToBeSent.length > 0 && <EmailBody email={findEmail(id)} body={body} markAsFav={markAsFav} />}
+        </DualPanel>
+      </div>
+    );
+  }
+
+  else {
+    return (
+      <h1>
+        Loading...
+      </h1>
+    )
+  }
+}
+
+export default App;
+
+//check if first render has occured to delay emailbody fetch
+function IsMounted() {
+  const isMountRef = useRef(false);
+  useEffect(() => {
+    isMountRef.current = true;
+  }, []);
+  return isMountRef.current;
+}
+
+//check if anything is stored in local storage
+function useLocalStorage(key, initialValue) {
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      return initialValue;
+    }
+  });
+
+  const setValue = value => {
+    try {
+      const valueToStore = value instanceof Function ? value(storedValue) : value;
+      setStoredValue(valueToStore);
+      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+    } catch (error) {
+      console.log(error);
+    }
+  };
+  return [storedValue, setValue];
+}
+```
+### Custom Hooks
+We have defined two custom hooks for various purposes. Let's understand what custom hooks are. Custom hooks enable the reuse of stateful logic by providing boundaries to maintain isolated states and side effects.
+
+The first custom hook called isMounted is used to check if the component is mounted or not. This is done to ensure performance. At the time of mounting, we want only the first useEffect to make the API call. All the other hooks should not be called. The lesser the number of API calls made, the faster the web page.
+
+The second custom hook is used to store the read and fav lists in local storage. This time, we consider it from the user experience perspective. Showing the user the list of reading and favorite, while reloading the page enhances the user experience. It is very important to store only relevant information. Therefore, the second custom hook takes in the input key and initialValue. It is checked if the key exists in the localStorage. If yes, it is updated with the new value, else a new key is created.
+
+To check the value of the localStorage, go to Developer Tools -> Application -> Local Storage. You will find the values stored there. Any change in the list will be immediately reflected in the localStorage.
+
+### Explaination: *Email.js*
+We use useContext to get the email. Each email is displayed with the incorporation of various styling components. The scroll component limits the display height and prevents overflow.  
```

Comment:
If you have followed my previous note about diving the reader context as to where that code should go. All you need to do here is say something along the lines of, "Just like we did for app.js you will now need to copy the following code into the email.js file previously mentioned in <provide section title>."